doc/classes/SpinBox.xml

```diff
@@ -13,23 +23,29 @@
 			<return type="LineEdit">
 			</return>
 			<description>
+				Returns the [Object] id of the [LineEdit] field of the [SpinBox]. You can use the [Object] id to invoke any [LineEdit] specific commands.
```

Comment:
It's not an object id, it is a reference (a pointer) to the LineEdit instance used by this SpinBox instance. It's the same thing as the `spin_box` variable in your above example, which points to the SpinBox instance.

An object ID is different, it's a number that represents the object's instance. It's not an object, and therefore you couldn't call any methods on it if it were an ID (like you can't call methods on other `int` values).

Reply:
![Screenshot_1](https://user-images.githubusercontent.com/1850856/60284420-2c1cee80-9914-11e9-8015-0dd9040e00d4.png)
Sorry, i might just be misunderstanding. Do you mean its a reference to the object id of the line-edit? in the debugger it says its an object id and also typeof() function says its an object. I'm confused.

Reply:
I can also invoke and LineEdit specific commands using that id.

Reply:
The debugger shows you that it's an Object, and gives you its ID. But that doesn't mean that the variable is an "Object ID", that's simply its representation in the debugger (the alternative would be to show the LineEdit itself in the debugger, but you can imagine this can quickly because tricky).

LineEdit *is* an Object, like all Nodes. Instanced objects have an Object or instance ID, accessed via `Object.get_instance_id()`, which gives you the identifier by which the engine keeps track of the instance.

So no, you're not using the ID here, you're using a reference to an instanced Object (of the subclass LineEdit, which inherits Object), which has the instance ID 1144.

Reply:
Cool! But:
![instance1](https://user-images.githubusercontent.com/1850856/60287989-ea904180-991b-11e9-868d-595b37376447.png)
![instance2](https://user-images.githubusercontent.com/1850856/60288000-ef54f580-991b-11e9-9964-dcd97979d0e3.png)

An instance ID ( which can be retrieved using Object.get_instance_id() as you stated) is a simple integer on which you can not invoke any commands. In this case, get_line_edit() method seems to be returning the object itself ( I'm not sure if i should say Object ID or just Object anymore) on which you can freely invoke any method inherited by LineEdit Class. So I think if we say that it returns the Instance ID instead of Object ID it will be misleading (because instance id is just an integer and you can not invoke any commands with it). Please advice.

Reply:
But it's not returning any **ID**. An ID is an integer. What you get here is not an ID, but an **instance** of an Object-derived class (here, a Node, and more specifically a LineEdit node).

I just mentioned the instance ID because this is what you **see** in the debugger as a simple representation of the Object instance pointer, but that's not what you actually get. When you do `get_node("LineEdit")`, you don't get the instance ID, you get the instance itself. Here, it's the same.

Reply:
I pushed an edit clarifying that (it's not that I didn't want to leave it to you, but I want to merge this PR asap before merging another one which will add default values to the classref, and thus introduce conflicts with any unmerged docs PR).